### PR TITLE
Proposal for threading

### PR DESCRIPTION
**Pull request by erik-megarad**

### CODE REVIEW

File: baserun/helpers.py [hunk 1]
```diff
@@ -14,4 +19,28 @@ class BaserunType(Enum):
 class BaserunStepType(Enum):
     LOG = auto()
     AUTO_LLM = auto()
-    CUSTOM_LLM = auto()
\ No newline at end of file
+    CUSTOM_LLM = auto()
+
+
+def baserun_threaded_wrapper(
+    target: Callable, results: list, thread_args=None, thread_kwargs=None
```
Comment (adam-baserun) [inline baserun/helpers.py:26]: Is it possible to leave the signature as:

`baserun_threaded_wrapper(target: Callable)` and basically give wrapper the args / kwargs parameters which will be passed through when invoked by the threading module?

Comment (adam-baserun) [inline baserun/helpers.py:26]: Is it possible for this target fn to be async? Seems generally weird to do that though

File: baserun/helpers.py [hunk 2]
```diff
@@ -14,4 +19,28 @@ class BaserunType(Enum):
 class BaserunStepType(Enum):
     LOG = auto()
     AUTO_LLM = auto()
-    CUSTOM_LLM = auto()
\ No newline at end of file
+    CUSTOM_LLM = auto()
+
+
+def baserun_threaded_wrapper(
```
Comment (adam-baserun) [inline baserun/helpers.py:25]: Consider moving this to baserun.py as a baserun.thread() or baserun.thread_wrapper() function  and expose via the __init__.py

Comment (adam-baserun) [inline baserun/helpers.py:25]: Should we bailout if baserun isn't initialized?

File: baserun/helpers.py (hunk 1 above)
Comment (erik-megarad) [reply baserun/helpers.py:26]: If they want it to be async they would need to make the target `asyncio.run` with their callable target as an arg. Essentially, nothing changes with the wrapper

**Pull request merged**